localtime/data_source_timestamp.go

```diff
@@ -68,10 +68,39 @@ func dataSourceTimestamp() *schema.Resource {
 }
 
 func dataSourceTimestampRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
-	// Warning or errors can be collected in a slice type
 	var diags diag.Diagnostics
 
-	// Add implementation details here
+	local_time := d.Get("local_time").(string)
+
+	layout := d.Get("layout").(string)
+	if layout == "" {
+		layout_local, err := dateparse.ParseFormat(local_time)
+		if err != nil {
+			return diag.FromErr(err)
```

Comment:
```suggestion
			return diag.Errorf("error parsing local time: %s", err)
```

Wrapping errors with context helps track them down, otherwise you're liable to return/print errors from `dateparse` that may be particularly hard to track down. Similar advice applies below for other errors.